Lib/test/test_pdb.py

```diff
@@ -1110,6 +1112,45 @@ def test_readrc_kwarg(self):
             if save_home is not None:
                 os.environ['HOME'] = save_home
 
+    def test_find_function_with_invalid_filename(self):
+        self.assertIsNone(pdb.find_function('foo', 'invalid filename'))
+
+    def test_getsourcelines(self):
+        def foo():
+            pass  # pragma: no cover
+        lines, lineno = inspect.findsource(foo)
+        expected_lines = inspect.getblock(lines[lineno:])
+        expected_lineno = lineno + 1
+        actual_lines, actual_lineno = pdb.getsourcelines(foo)
+        self.assertEqual(actual_lines, expected_lines)
+        self.assertEqual(actual_lineno, expected_lineno)
+
+    def test_getsourcelines_with_module_frame_obj(self):
+        for frame_tuple in inspect.getouterframes(inspect.currentframe()):
+            frame = frame_tuple[0]
+            if frame.f_globals is frame.f_locals:
+                module_frame = frame
+                break
```

Comment:
I don't like this hack to get frame object for module...
But I don't know is there any way.
I'm not expert of inspect and pdb.